hooks/gate-scripts/lib/marker_check.py

```
@@ -2089,6 +2089,15 @@ def _glob_helper_targeted(word, deep=None):
     return _glob_helper(word, deep)
 
 
+# Whole-token POSIX/bash digit-negation globs (#776). Optional stars either side.
+_DIGIT_NEGATION_ONLY_RE = re.compile(r"^\*?\[(?:!|\^)0-9\]\*?$")
+
+
+def _is_digit_negation_only_segment(seg):
```

Ruff (0.16.2)

[warning] Missing return type annotation for private function `_is_digit_negation_only_segment`

(ANN202)